### ISSUE TITLE
additional snapshot version component placeholder

### ISSUE THREAD

**Issue by olibraga** [qoomon/maven-git-versioning-extension#182]
Would be nice to have a version.snapshot placeholder having value -SNAPSHOT if ${version} ends in -SNAPSHOT, blank otherwise.
This would allow easily generating version 1.2.3-generated-part-SNAPSHOT instead of 1.2.3-SNAPSHOT-generated-part from  1.2.3-SNAPSHOT

**Comment by qoomon** [comment]
Why not just `${version.release}-generated-part-SNAPHOT`

**Comment by olibraga** [comment]
Then I would get a trailing -SNAPSHOT whether the pom version contained -SNAPSHOT or not

**Comment by qoomon** [comment]
Understand, however the idea of this is extension to generate versions based on branches and tags so best practice is to always have snapshot version in pom.xml. I'm not sure I this feature request make sense for this extension.

Could you explain your specific usecase/workflow?

**Comment by olibraga** [comment]
The "generated-part" above is supposed to stand for whatever the extension adds to the version, e.g. "feature-appkfk-3573-expand-customer-api", resulting in say "appkfk-1.2.3-feature-appkfk-3573-expand-customer-api-SNAPSHOT.jar".

I want to be able to specify a rule that adds -SNAPSHOT to the end of the version if the developer specified a -SNAPSHOT version in the pom, so a "appkfk-1.2.3-hotfix-appkfk-3574-getcustomer-broken-whe-SNAPSHOT.jar" is generated instead of "appkfk-1.2.3-SNAPSHOT-hotfix-appkfk-3574-getcustomer-broken-whe.jar"

**Comment by qoomon** [comment]
I recommend following extension configuration. Would that solve your issue?
```xml
<ref type="branch">
    <pattern>main</pattern>
    <version>${version}</version>
</ref>
<ref type="branch">
    <pattern><![CDATA[feature/(?<feature>.+)]]></pattern>
    <version>${version.release}-${ref.feature}-SNAPSHOT</version>
</ref>
```

**Comment by qoomon** [comment]
However I could add a placeholder. I'm just not sure about the name and wording yet. I there a better wording for `.prefixed`?
* `${version.label}`  split on first dash e.g. `SNAPSHOT`, `FOO-BAR`
* `${version.label.prefixed}`  e.g. `-SNAPSHOT`, `-FOO-BAR`

* `${version.release}` - remove label  e.g. `1.3.2`
* `${version.major}`  e.g. `1`
* `${version.minor}`  e.g. `3`
* `${version.minor.prefixed}`  e.g. `.3`
* `${version.patch}`  e.g. `2`
* `${version.patch.prefixed}`   e.g. `.2`

**Comment by olibraga** [comment]
Your ${version.label} and .prefixed change would solve my problem in a more general way than my ${version.snapshot} suggestion, great!

P.S. Your sample refs above do not solve my problem, I did not do a good job of explaining it.

**Comment by qoomon** [comment]
Okay I'll add these place holders.

**Comment by qoomon** [comment]
all placeholders are available in version `7.4.0`
